src/lib.rs

```diff
@@ -168,12 +156,9 @@ impl io::Read for EventedAnonRead {
             if !self.consumer.is_empty() {
                 self.inner.readiness.set_readiness(Ready::readable())?;
             }
-
-            if !self.consumer.is_full() {
-                self.inner.sig_buffer_not_full.notify_one();
-            }
         }
 
+        self.inner.sig_buffer_not_full.notify_one();
```

Comment:
Would it hurt to `sig_buffer_not_full.notify_one()` if the buffer is in fact full? Could it happen?

Reply:
I don't think it'll cause any problems, the worker thread will just make one turn of the inner loop and then block again.